src/Gold.Redis/Gold.Redis.LowLevelClient/Communication/RedisSocketAuthenticator.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Gold.Redis.Common.Interfaces.Communication;
+using System;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+
+namespace Gold.Redis.LowLevelClient.Communication
+{
+    public class RedisSocketAuthenticator : IRedisAuthenticator
+    {
+        private readonly ISocketCommandExecutor _socketCommandExecutor;
+        public RedisSocketAuthenticator(ISocketCommandExecutor socketCommandExecutor)
+        {
+            _socketCommandExecutor = socketCommandExecutor;
+        }
+        public async Task<bool> TryAuthenticate(Socket connectionSocket, string password)

```

Comment:
I am still not 100% about this class.
first of all it's not testable. it will be really hard to moq a socket. 
also why cant it be a function in SocketsConnectionsContainer? the redis auth mechanism will probably never change, and it will probably never be used somewhere else then SocketsConnectionsContainer.

Reply:
SRP -> its not the sokets container job to know how to authenticate to redis + 
Redis can add other options to do log in (Oath \ diffrent creds) we still dont know about so I think this is a valid class & interface

Reply:
I would think that the function called 'Connect' will know how to handle a simple matter like auth.
I really think that when the day comes and redis adds more auth methods it will be not so difficult to modify the code to work with that. the additional complexity a class like this adds is really un-necessary in the time being

Reply:
its make the code more readable and more solid. I really don't think its causing over design or over complication  of any sort. its only adds one more class + interface that keeps the origin class with no need to know the AUTH command + keeps its below 100 lines of code. so i really do think its necessary